cmd/snap/cmd_info.go

```diff
@@ -163,27 +164,78 @@ func coalesce(snaps ...*client.Snap) *client.Snap {
 	return nil
 }
 
-// formatDescr formats a given string (typically a snap description)
+// runesTrimRightSpace returns text, with any trailing whitespace dropped.
+func runesTrimRightSpace(text []rune) []rune {
+	j := len(text)
+	for j > 0 && unicode.IsSpace(text[j-1]) {
+		j--
+	}
+	return text[:j]
+}
+
+// runesLastIndexSpace returns the index of the last whitespace rune
+// in the text. If the text has no whitespace, returns -1.
+func runesLastIndexSpace(text []rune) int {
+	for i := len(text) - 1; i >= 0; i-- {
+		if unicode.IsSpace(text[i]) {
+			return i
+		}
+	}
+	return -1
+}
+
+// wrapLine wraps a line to fit into width, preserving the line's indent, and
+// writes it out prepending padding to each line.
+func wrapLine(out io.Writer, text []rune, pad string, width int) {
+	// Note: this is _wrong_ for much of unicode (because the width of a
+	//       rune on the terminal is anything between 0 and 2, not always 1
+	//       as this code assumes) but fixing that is Hard. Long story
+	//       short, you can get close using a couple of big unicode tables
+	//       (which is what wcwidth does). Getting it 100% requires a
+	//       terminfo-alike of unicode behaviour.
+	// This (and possibly printDescr below) should move to strutil once
+	// we're happy with it getting wider (heh heh) use.
+
+	// discard any trailing whitespace
+	text = runesTrimRightSpace(text)
+	// establish the indent of the whole block
+	idx := 0
+	for idx < len(text) && unicode.IsSpace(text[idx]) {
+		idx++
+	}
+	indent := pad + string(text[:idx])
+	text = text[idx:]
+	width -= idx + utf8.RuneCountInString(pad)
+	for len(text) > width {
+		// find a good place to chop the text
+		idx = runesLastIndexSpace(text[:width+1])
+		if idx < 0 {
+			// there's no whitespace; just chop at line width
+			idx = width
+		}
+		fmt.Fprint(out, indent, string(text[:idx]), "\n")
```

Comment:
This function needs to account for errors if it takes an io.Writer as an input, since we don't know what writer that might be (a buffer, local file, socket, ...). If dropping errors is desired, that's a responsibility of the call site, not of a formatting function. Alternatively, we need to change the function so it handles memory only, in and out. With bytes.Buffer, for example.